// library/InputField.js
/* 
		WARNING 
		Auto generated file. 
		Do not modify its contents.
*/

const extend = require('js-base/core/extend');
const TextBox = require('sf-core/ui/textbox');
const FlexLayout = require('sf-core/ui/flexlayout');
const Color = require('sf-core/ui/color');
const Font = require('sf-core/ui/font');
const ImageView = require('sf-core/ui/imageview');
const Image = require('sf-core/ui/image');
const ImageFillType = require('sf-core/ui/imagefilltype');



const getCombinedStyle = require("library/styler-builder").getCombinedStyle;

const InputField = extend(FlexLayout)(
	//constructor
	function(_super, props) {
		// initalizes super class for this component scope
		_super(this, props);

		var tbInput = new TextBox(getCombinedStyle(".textBox", {
			left: 0,
			top: 0,
			width: 375,
			height: 60,
			text: "Input",
			right: 0,
			bottom: 0,
			positionType: FlexLayout.PositionType.ABSOLUTE,
			backgroundColor: Color.create(0, 255, 255, 255),
			font: Font.create("Arial", 32, Font.NORMAL)
		})); 
		this.addChild(tbInput);
		this.tbInput = tbInput;

		var imgLine = new ImageView(getCombinedStyle(".imageView", {
			width: NaN,
			height: 2,
			left: 0,
			image: Image.createFromFile("images://input_active_line.png"),
			imageFillType: ImageFillType.STRETCH,
			positionType: FlexLayout.PositionType.ABSOLUTE,
			right: 0,
			bottom: 0,
			backgroundColor: Color.create(0, 255, 255, 255)
		})); 
		this.addChild(imgLine);
		this.imgLine = imgLine;

		//assign the children to inputField 
		this.children = Object.assign({}, {
			tbInput: tbInput,
			imgLine: imgLine
		});
		
	});

InputField.defaults = getCombinedStyle(".flexLayout", {
	left: 0,
	top: 0,
	width: 375,
	height: 60,
	right: 0,
	positionType: FlexLayout.PositionType.RELATIVE,
	backgroundColor: Color.create(0, 255, 255, 255)
});

module && (module.exports = InputField);
